### PR TITLE
[move][bytecode verifier] Restructure dependency verifications

### PR DESCRIPTION
**Pull request by tnowacki**
- Cleanup for dependency checker
- Should make new visibility checks easier

## Motivation

- The code had a lot of function parameters that were not needed, and the generation of this "checker" object was very brittle 

## Test Plan

- ran tests

## Related PRs

Conflicts with #7399

### CODE REVIEW

File: language/bytecode-verifier/src/dependencies.rs [hunk 1]
```diff
@@ -1,415 +1,341 @@
...
-                self.struct_id_to_handle_map.insert(
+                checker.struct_id_to_handle_map.insert(
                     (module_id.clone(), struct_name.to_owned()),
                     struct_def.struct_handle,
                 );
             }
             // Module::FuncName -> def handle idx
             for func_def in module.function_defs() {
-                if !func_def.is_public {
+                if !func_def.is_public() {
```
Comment (mengxu-fb) [inline language/bytecode-verifier/src/dependencies.rs:74]: I guess this might conflict with #7418 as well.

Comment (tnowacki) [reply language/bytecode-verifier/src/dependencies.rs:74]: It will, yeah

File: language/bytecode-verifier/src/dependencies.rs [hunk 2]
```diff
@@ -109,289 +76,266 @@ impl<'a> DependencyChecker<'a> {
...
+
+fn verify_module_impl<'a>(
+    module: &CompiledModule,
+    dependencies: impl IntoIterator<Item = &'a CompiledModule>,
+) -> PartialVMResult<()> {
+    let checker = &DependencyChecker::module(module, dependencies);
+
+    verify_imported_modules(checker)?;
+    verify_imported_structs(checker)?;
+    verify_imported_functions(checker)
```
Comment (mengxu-fb) [inline language/bytecode-verifier/src/dependencies.rs:104]: Would it be better to move these `verify_imported_*` functions into `DependencyChecker`? So we can do
`checker.verify_imported_modules()?;` ...

**Review by mengxu-fb — APPROVED** [review]
LGTM!

File: language/bytecode-verifier/src/dependencies.rs (hunk 2 above)
Comment (tnowacki) [reply language/bytecode-verifier/src/dependencies.rs:104]: No. These aren't operations belonging to checker. The checker is just a context under which these checks are performed. 

So there is an intentional API separation there (this is why I hate the checker code that exists today, all of the logic is all bundled together and there is no clear split between context of the check and the checks themeslves)

I'll probably rename all of these in this file to maybe alleviate confusion

**Pull request merged**